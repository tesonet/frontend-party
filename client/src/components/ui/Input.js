import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { InputStyle } from './style';

class Input extends Component {
  static propTypes = {
    type: PropTypes.string,
    placeholder: PropTypes.string,
    icon: PropTypes.string,
    onChange: PropTypes.func
  };

  static defaultProps = {
    type: 'text',
    placeholder: '',
    icon: '',
    onChange: () => {}
  };

  constructor(props) {
    super(props);
    this.state = {
      value: '',
      focus: false
    }
  }

  onChange = (event) => {
    const { onChange } = this.props;
    const { value } = event.target;

    onChange(value);
    this.setState({ value });
  }

  toggleFocus() {
    const { focus } = this.state;
    this.setState({ focus: !focus });
  }

  renderIcon() {
    const { icon } = this.props;

    return (
      <i className={`fa fa-${icon}`}></i>
    );
  }

  render() {
    const { type, placeholder, value, icon, invalid } = this.props;
    const state = this.state;
    const { focus } = state;
    const showIcon = icon && icon !== '' && state.value === '' && !focus;
    let className = ['input'];
    showIcon && className.push('icon');
    invalid && className.push('invalid');

    return (
      <div className={'input-wrapper'}>
        {showIcon && this.renderIcon()}
        <input
          type={type}
          value={value}
          placeholder={placeholder}
          onChange={this.onChange}
          className={className.join(' ')}
          onFocus={() => this.toggleFocus()}
          onBlur={() => this.toggleFocus()}
        />
      </div>
    );
  }
}

export default Input;
