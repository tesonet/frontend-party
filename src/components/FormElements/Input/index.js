import InputComposer from './InputComposer';
import Text from './Text';

export const Input = {
  Text: InputComposer(Text)
};
