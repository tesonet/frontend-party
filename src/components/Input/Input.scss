.input-group {
    .input-group-text {
        background-color: #ffffff;
        border: none;
    }
    .form-control {
        border: none;
        outline: none;
    }
}