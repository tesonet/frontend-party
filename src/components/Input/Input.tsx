import * as React from 'react';
import styled from 'styled-components';
import { themeGet } from '@styled-system/theme-get';
import { useField } from 'formik';

import { ErrorMessage } from './ErrorMessage/ErrorMessage';

interface InputProps {
  id: string;
  name: string;
  placeholder: string;
  type: 'text' | 'password';
  icon?: React.ReactNode;
}

interface StyledInputProps extends InputProps {
  hasIcon: boolean;
  hasError: boolean;
}

const InputWrapper = styled.div`
  position: relative;

  svg {
    position: absolute;
    left: 1.8rem;
    top: 50%;
    transform: translateY(-50%);
  }
`;

const StyledInput = styled.input<StyledInputProps>`
  color: ${themeGet('colors.text')};
  height: 5.8rem;
  width: 100%;
  border-radius: 6px;
  padding-left: ${({ hasIcon }) => (hasIcon ? '5.6rem' : '1.8rem')};
  border-width: ${({ hasError }) => (hasError ? '2px' : '0')};
  border-color: ${({ hasError }) => (hasError ? themeGet('colors.error') : 'none')};
  font-weight: ${themeGet('fontWeights.lighter')};
  padding-top: 5px;

  &::placeholder {
    color: ${themeGet('colors.lightText')};
  }
`;

export const Input = ({ icon, name, ...rest }: InputProps) => {
  const [field, meta] = useField({ name, ...rest });

  const { error, touched } = meta;

  return (
    <>
      <InputWrapper>
        <StyledInput hasIcon={!!icon} hasError={!!error && touched} {...field} {...rest} />
        {icon && icon}
      </InputWrapper>
      <ErrorMessage>{!!error && touched && error}</ErrorMessage>
    </>
  );
};
