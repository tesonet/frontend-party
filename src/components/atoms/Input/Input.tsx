import React from 'react';
import styles from './Input.module.scss';

type Falsy = false | 0 | '' | null | undefined;

interface IInputProps {
	type: string;
	name?: string;
	startAdornment?: React.ReactNode;
	endAdornment?: React.ReactNode;
	placeholder?: string;
	errorMessage?: string | Falsy;
	inputStyles?: string;
	onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

export function Input(props: IInputProps) {
	return (
		<div className={styles['input-wrapper']}>
			{
				(props.startAdornment || props.endAdornment) &&
                <span className={[
					styles['input__adornment'],
					props.startAdornment && styles['input__adornment--start'],
					props.endAdornment && styles['input__adornment--end'],
				].filter(Boolean).join(' ')}>
					{props.startAdornment || props.endAdornment}
				</span>
			}
			<input
				name={props.name || ''}
				type={props.type || 'text'}
				className={[
					styles['input'],
					props.errorMessage && styles['input--error'],
					props.inputStyles,
				].filter(Boolean).join(' ')}
				placeholder={props.placeholder || ''}
				onChange={e => props.onChange && props.onChange(e)}
			/>
			{
				props.errorMessage &&
                <span className={styles['input__error']}>
					{props.errorMessage}
				</span>
			}
		</div>
	);
}