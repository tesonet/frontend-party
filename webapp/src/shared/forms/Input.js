import React from 'react';

const defaultProps = {
  type: 'text',
  name: '',
  value: '',
  placeholder: '',
  icon: null,
  error: null,
  onChange: () => {}
};

export default function Input(props) {
  return (
    <div className="form-group with-icon">
      <input
        type={ props.type }
        className={ `form-control ${props.error ? 'is-invalid' : ''}` }
        name={ props.name }
        value={ props.value }
        onChange={ props.onChange }
        placeholder={ props.placeholder }
      />

      <div className="icon">
        { props.icon }
      </div>

      <div className="invalid-feedback">
        { props.error }
      </div>
    </div>
  );
}

Input.defaultProps = defaultProps;
