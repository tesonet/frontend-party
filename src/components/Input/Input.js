import React from 'react';
import PropTypes from 'prop-types';
import './Input.css';
import ReactSVG from 'react-svg';

const input = ({ id, label, type, icon, value, onChange }) => {
    let svg = null;
    if (icon) {
        switch (icon) {
            case 'user':
                svg = <ReactSVG path="assets/svg/user.svg" className="icon"/>;
                break;
            case 'lock':
                svg = <ReactSVG path="assets/svg/lock.svg" className="icon"/>;
                break;
        }
    }
    const prepend = svg ? (
        <div className="input-group-prepend">
            <span className="input-group-text">{svg}</span>
        </div>
    ) : '';
    
    return (
        <div className="input-group mb-3">
            {prepend}
            <input
                id={id}
                type={type ? type : 'text'}
                className="form-control"
                placeholder={label}
                aria-label={label}
                value={value ? value : ''}
                autoComplete="false"
                onChange={onChange}
            />
        </div>
    );
};

input.propTypes = {
    label: PropTypes.string
};

export default input;