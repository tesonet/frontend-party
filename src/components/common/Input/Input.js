import React from "react";

import "./Input.scss";
import {joinTruthy} from "../../../utils/utils";

export const Input = (props) => {
  return (
    <div className={joinTruthy([
      "input",
      props.classNames?.wrapper,
    ])}>
      <input
        placeholder={props.placeholder}
        className={joinTruthy([
          "input__input-field",
          props.classNames?.inputField,
        ])}
      />
      {props.errors && (
        <span className="input__error-message" >
          {props.errors}
        </span>
      )}
    </div>
  );
};

export default Input;