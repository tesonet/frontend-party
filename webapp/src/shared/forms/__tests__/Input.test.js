import React from 'react';
import Input from '../Input';

it('renders without crashing', () => {
  const wrapper = shallow(<Input />);
  expect(wrapper).toHaveLength(1);
});

it('has reasonable defaultProps', () => {
  expect(Input.defaultProps).toEqual({
    type: 'text',
    name: '',
    value: '',
    placeholder: '',
    icon: null,
    error: null,
    onChange: expect.any(Function)
  });
});

it('matches the snapshot', () => {
  const wrapper = shallow(<Input
    error="An error occurred"
    name="username"
    value=""
    placeholder="Username"
    icon={ <span className="oi oi-person" /> }
  />);

  expect(wrapper).toMatchSnapshot();
});
