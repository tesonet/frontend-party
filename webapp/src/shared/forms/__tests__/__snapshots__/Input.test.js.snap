// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="form-group with-icon"
>
  <input
    className="form-control is-invalid"
    name="username"
    onChange={[Function]}
    placeholder="Username"
    type="text"
    value=""
  />
  <div
    className="icon"
  >
    <span
      className="oi oi-person"
    />
  </div>
  <div
    className="invalid-feedback"
  >
    An error occurred
  </div>
</div>
`;
